crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -4583,9 +4583,10 @@ impl<'db> TypeInferenceBuilder<'db> {
 
     fn infer_parameterized_known_instance_type_expression(
         &mut self,
+        subscript: &ast::ExprSubscript,
         known_instance: KnownInstanceType,
-        parameters: &ast::Expr,
     ) -> Type<'db> {
+        let parameters = &subscript.slice;
```

Comment:
if you do this then you can avoid the `.as_ref()` call on line 4611:

```suggestion
        let parameters = &*subscript.slice;
```

Reply:
Great I did not know about this.